### PR TITLE
Refine Script and args for Cpp Graph

### PR DESCRIPTION
**Pull request by zhenwei-intel**
## Type of Change

Script change
1. Convert
           python scripts/convert.py --outtype f32 --outfile ne-f32.bin model_path/model_id
2. Quant

            python scripts/quant.py --model_name llama --model_file ne-f32.bin --out_file ne-q4_0.bin --bits 4   

3. Inference

            python scripts/inference.py --model_name llama -m ne-q4_j.bin -c 512 -b 1024 -n 256 -t 56 --color -p "She opened the door and see“

4. Unify as ONE WIP

            python scripts/run_llm.py

5. Args change
          Sync with INC block_size   =>    group_size

## Description

detail description 
JIRA ticket: xxx

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 1]
```diff
@@ -89,7 +89,7 @@ quantization args explanations:
 | --block_size    | block size (default: 32)                                    |
 | --scale_dtype   | fp32/bf16 type for scales (default: fp32)                   |
 | --compute_type  | Gemm computation data type: int8/fp32/ggml (default: ggml)  |
-
+| --build_dir     | build directory of binary executable files                  |
```
Comment (kevinintel) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:92]: can we combine model_file and build_dir ?

**Review by a32543254 — APPROVED** [review]
LGTM

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 2]
```diff
@@ -123,7 +123,7 @@ LLM running script args explanations:
 | --color           | colorise output to distinguish prompt and user input from generations   |
 | --keep            | number of tokens to keep from the initial prompt (default: 0, -1 = all) |
 | --glm_tokenizer   | the path of the chatglm tokenizer (default: THUDM/chatglm-6b)           |
-
+| --build_dir       | build directory of binary executable files                              |
```
Comment (kevinintel) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:126]: can we combine model and build_dir?

File: intel_extension_for_transformers/llm/runtime/graph/scripts/one_click_run.py [hunk 3]
```diff
@@ -19,11 +19,16 @@
...
 import subprocess
 
+build_path = Path(Path(__file__).parent.absolute(), "../build/")
+
 def main(args_in: Optional[List[str]] = None) -> None:
     parser = argparse.ArgumentParser(description="run quantization and inference")
     parser.add_argument(
         "model", type=Path, help="directory containing model file or model id"
     )
+    parser.add_argument(
```
Comment (kevinintel) [inline intel_extension_for_transformers/llm/runtime/graph/scripts/one_click_run.py:29]: one click may not need this, but we should show logs when build folder is not exist

File: intel_extension_for_transformers/llm/runtime/graph/README.md (hunk 1 above)
Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/README.md:92]: model_file is path to model bin, different from build directory

File: intel_extension_for_transformers/llm/runtime/graph/scripts/one_click_run.py (hunk 3 above)
Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/scripts/one_click_run.py:29]: we have logs 
```
    if not path.exists():
        print(path)
        print("Please build graph first or select the correct model name.")
```
one click script will call quantize.py and inference.py, so if user has user-defined build directory, user should call one-click script with build_dir path

**Comment by luoyu-intel** [comment]
@hshen14 renaming block_size to group_size

**Review by zhentaoyu — APPROVED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/application/common.cpp [hunk 4]
```diff
@@ -679,7 +679,7 @@ void quant_print_usage(int argc, char** argv, const quant_params& params) {
   fprintf(stderr, "  --nthread N           number of threads to use (default: 1)\n");
   fprintf(stderr, "  --bits N              number of bits to use for quantization (default: 4)\n");
```
Comment (luoyu-intel) [inline intel_extension_for_transformers/llm/runtime/graph/application/common.cpp:680]: py parameter "weight_dtype" and cpp parameter "bits" should be the same

**Review by luoyu-intel — CHANGES_REQUESTED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 5]
```diff
@@ -116,14 +116,14 @@ LLM running script args explanations:
...
 | -n / --n_predict  | number of tokens to predict (default: -1, -1 = infinity)                |
 | -t / --threads    | number of threads to use during computation (default: 56)               |
-| -b / --batch_size | batch size for prompt processing (default: 512)                         |
+| -b / --batch_size_truncate | batch size for prompt processing (default: 512)                         |
 | -c / --ctx_size   | size of the prompt context (default: 512, can not be larger than specific model's context window length)                                                                                |
 | -s / --seed       | NG seed (default: -1, use random seed for < 0)                          |
 | --repeat_penalty  | penalize repeat sequence of tokens (default: 1.1, 1.0 = disabled)       |
 | --color           | colorise output to distinguish prompt and user input from generations   |
 | --keep            | number of tokens to keep from the initial prompt (default: 0, -1 = all) |
 | --glm_tokenizer   | the path of the chatglm tokenizer (default: THUDM/chatglm-6b)           |
```
Comment (hshen14) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:125]: should be tokenizer? more general...

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/README.md:125]: this tokenizer is specially designed for Chatglm1 by zhenzhong

File: intel_extension_for_transformers/llm/runtime/graph/application/common.cpp (hunk 4 above)
Comment (luoyu-intel) [reply intel_extension_for_transformers/llm/runtime/graph/application/common.cpp:680]: Users complained about the parameter difference between Python and cpp.

Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/application/common.cpp:680]: @kevinintel @a32543254

Comment (a32543254) [reply intel_extension_for_transformers/llm/runtime/graph/application/common.cpp:680]: how about we change cpp bits to weight_dtype? 
what do you think of it ?

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 6]
```diff
@@ -116,14 +116,14 @@ LLM running script args explanations:
 | -p / --prompt     | prompt to start generation with (default: empty)                        |
 | -n / --n_predict  | number of tokens to predict (default: -1, -1 = infinity)                |
 | -t / --threads    | number of threads to use during computation (default: 56)               |
-| -b / --batch_size | batch size for prompt processing (default: 512)                         |
+| -b / --batch_size_truncate | batch size for prompt processing (default: 512)                         |
```
Comment (hshen14) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:119]: is it truncate for batch size or prompt length?

Comment (a32543254) [reply intel_extension_for_transformers/llm/runtime/graph/README.md:119]: truncate for prompt length

**Review by airMeng — APPROVED** [review]

**Review by hshen14 — APPROVED** [review]

File: intel_extension_for_transformers/llm/runtime/graph/README.md [hunk 7]
```diff
@@ -60,21 +60,21 @@ LLM Runtime assumes the same model format as [llama.cpp](https://github.com/gger
...
-# to neuarl engine graph optimized q4_j with 128 block_size format (recommended)
-python scripts/quant_bin.py --model_name llama2 --model_file ne-f32.bin --out_file ne-q4_j.bin --weight_dtype int4 --block_size 128 --compute_type int8
+# to neural engine graph optimized q4_j with 128 group_size format (recommended)
+python scripts/quantize.py --model_name llama2 --model_file ne-f32.bin --out_file ne-q4_j.bin --weight_dtype int4 --group_size 128 --compute_type int8
 
 # to ggml q4_0 format
-python scripts/quant_bin.py --model_name llama2 --model_file ne-f32.bin --out_file ne-q4_0.bin --weight_dtype int4
-# to neuarl engine graph optimized q4_j with 32 block_size format
+python scripts/quantize.py --model_name llama2 --model_file ne-f32.bin --out_file ne-q4_0.bin --weight_dtype int4
+# to neural engine graph optimized q4_j with 32 group_size format
```
Comment (hshen14) [inline intel_extension_for_transformers/llm/runtime/graph/README.md:75]: how about unifying to LLM runtime? seems we are still using neural engine ...

Comment (hshen14) [reply intel_extension_for_transformers/llm/runtime/graph/README.md:75]: or just make it simple: optimized INT4 model with group size 32

**Comment by VincyZhang** [comment]
cpp graph test: https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/93/

File: intel_extension_for_transformers/llm/runtime/graph/README.md (hunk 7 above)
Comment (zhenwei-intel) [reply intel_extension_for_transformers/llm/runtime/graph/README.md]: done

**Comment by VincyZhang** [comment]
https://inteltf-jenk.sh.intel.com/view/nlp-toolkit-validation/job/ITREX-cpp-graph-extension/94/

**Pull request merged**